barq/main.py

```diff
@@ -187,27 +163,18 @@ def menu_forward(menu):
         main_loop()
 
 
-def menu_backward():
+def menu_backward() -> None:
     """
-    Go back to previous menu (Pull from menu stack)
+    Go back to main menu 
     :return: None
     """
-    global menu_stack
-    try:
-        current_menu = menu_stack.pop()
-        next_menu = menu_stack[-1]
-        if next_menu == 'main':
-            go_to_menu(next_menu)
-        elif next_menu == 'training':
-            go_to_menu(next_menu)
-        elif next_menu == 'ec2instances':
-            go_to_menu(next_menu)
-    except Exception as e:
-        print(e)
-        pass
```

Comment:
always navigate back to main menu as the stack has only 2 cases 
- ['main', 'training]
- ['main', 'ec2instances']

so, menu_stack.pop() removes the last element in both cases return to main. However, dynamic navigation should be written based on stack complexity and supported features.